modules/gcloud/src/main/java/org/testcontainers/containers/GCloudGenericContainer.java

```diff
@@ -0,0 +1,66 @@
+package org.testcontainers.containers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.images.builder.dockerfile.DockerfileBuilder;
+
+/**
+ * @author Eddú Meléndez
+ */
+public class GCloudGenericContainer<SELF extends GCloudGenericContainer<SELF>> extends GenericContainer<SELF> {
+
+	public static final String DEFAULT_GCLOUD_IMAGE = "google/cloud-sdk:291.0.0-alpine";
+
+	private List<String> commands = new ArrayList<>();
+
+	public GCloudGenericContainer(String image) {
+		super(image);
+	}
+
+	public GCloudGenericContainer(String image, String mainCmd, String[] prerequisiteCmds) {
+		super(buildImage(image, mainCmd, prerequisiteCmds));
+	}
+
+	@Override
+	protected void containerIsStarted(InspectContainerResponse containerInfo) {
+		runAdditionalCommands();
+	}
+
+	private void runAdditionalCommands() {
+		this.commands.forEach(cmd -> {
+			try {
+				execInContainer(cmd);
+			} catch (IOException | InterruptedException e) {
+				logger().error("Failed to execute {}. Exception message: {}", cmd, e.getMessage());
+			}
+		});
+	}
+
+	public SELF withAdditionalCommands(String... cmds) {
+		this.commands.addAll(Arrays.asList(cmds));
+		return self();
+	}
+
+	private static ImageFromDockerfile buildImage(String image, String mainCmd, String[] prerequisiteCmds) {
```

Comment:
does it really need to be an image? Can't we just use the command for it, since the image is not cached anyways?

Reply:
At the beginning I was not able to run it using command but that led me to know more about `testcontainers`. Now, I did it but had to set up a `startupTimeout`.

Reply:
Are we sure about this? 😄 
It looks like these setup steps will require a non-trivial amount of time to execute - perhaps even the majority of the startup time.

Would it be worth trying to get more attention on https://github.com/GoogleCloudPlatform/cloud-sdk-docker/issues/202 so that we can avoid having these setup steps entirely?

Reply:
Actually, I missed that we install Java and other things... Yeah, I believe we need to wait for these images :(

Reply:
/cc @jamesward @saturnism